src/darwin/Framework/CHIPTests/MTRAsyncCallbackQueueTests.m

```diff
@@ -256,4 +256,160 @@ - (void)testInvalidation
     [self waitForExpectations:@[ expectation, cancelExpectation ] timeout:5];
 }
 
+- (void)testBatching
+{
+    XCTestExpectation * workItem1ReadyExpectation = [self expectationWithDescription:@"Work item 1 called"];
+    __block BOOL workItem2BatchingCalled = NO;
+    __block BOOL workItem2ReadyCalled = NO;
+    XCTestExpectation * workItem3ReadyExpectation = [self expectationWithDescription:@"Work item 3 called"];
+
+    MTRAsyncCallbackWorkQueue * workQueue = [[MTRAsyncCallbackWorkQueue alloc] initWithContext:nil queue:dispatch_get_main_queue()];
+
+    // Have a work item sleep so the testing items can queue
+    MTRAsyncCallbackQueueWorkItem * workItem0 =
+        [[MTRAsyncCallbackQueueWorkItem alloc] initWithQueue:dispatch_get_global_queue(QOS_CLASS_DEFAULT, 0)];
+    MTRAsyncCallbackReadyHandler readyHandler0 = ^(MTRDevice * _Nonnull device, NSUInteger retryCount) {
+        // Give the following code a chance to enqueue and merge
+        sleep(1);
```

Comment:
Alternately, could we just call a function here that adds the other items, instead of sleeping?  Might be more resistant to race issues.

Reply:
Done.